chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1326,9 +1314,18 @@ impl EthereumAdapterTrait for EthereumAdapter {
                 } else {
                     // Decode failures are reverts. The reasoning is that if Solidity fails to
                     // decode an argument, that's a revert, so the same goes for the output.
-                    call.function.decode_output(&output).map_err(|e| {
+                    let result = call.function.decode_output(&output).map_err(|e| {
                         EthereumContractCallError::Revert(format!("failed to decode output: {}", e))
-                    })
+                    });
+                    let _ = graph::spawn_blocking_allow_panic(move || {
```

Comment:
I think I could move it to dedicated and_then future using https://docs.rs/futures-async-combinators/latest/futures_async_combinators/future/fn.and_then.html
But I didn't see it usage in the graph so I'm hesitant to add this